Documentation/Profiling/IL Rewriting Basics.md

```diff
@@ -0,0 +1,23 @@
+# IL Rewriting Basics
+
+## Intro
+One of the common use cases of the `ICorProfiler*` interfaces is to perform IL rewriting. Some possible reasons a profiler would want to rewrite IL:
+- Inspecting interesting process state
+  - Capturing exception state
+  - Inspecting managed objects
+  - Inspecting function arguments/return values
+- Injecting method hooks at the start/end of the method that call in to another managed library
+
+There are two ways to rewrite IL
+
+1. At Module load time with `ICorProfilerInfo::SetILFunctionBody`
+    This approach has the benefit that it is 'set it and forget it'. You can replace the IL at module load, and the runtime will treat this new IL as if the module contained that IL - you don't have to worry about any of the quirks of ReJIT. The downside is that is is unrevertable - once it is set, you cannot change your mind.
+
+2. At any point during the process lifetime with 'ICorProfilerInfo4::RequestReJIT' or 'ICorProfilerInfo10::RequestReJITWithInliners'.
+   This approach means that you can modify functions in response to changing conditions, and you can revert the modified code if you decide you are done with it. See the other entries about ReJIT in this folder for more information.
+
+## How to rewrite IL
+Hopefully this section will be fleshed out in the future. Right now we have some documentation in the archives at [Creating an IL-rewriting profiler](davbr-blog-archive/Creating an IL-rewriting profiler.md), but there is no start to finish tutorial on IL rewriting.
```

Comment:
GitHub isn't rendering/detecting the relative link to 'davbr-blog-archive/Creating an IL-rewriting profiler.md' properly? 

Maybe there's a way to get it working, but I suspect just changing it to the full URL might be easier/quicker!

Reply:
It's not github, as written it's a invalid link destination according to the [CommonMark-spec](https://spec.commonmark.org/0.29/#link-destination) due the whitespaces in the url.

So the need to be "escaped".
```suggestion
Hopefully this section will be fleshed out in the future. Right now we have some documentation in the archives at [Creating an IL-rewriting profiler](<./davbr-blog-archive/Creating an IL-rewriting profiler.md>), but there is no start to finish tutorial on IL rewriting.
```

Reply:
FWIW `%20` for spaces also works

---
In reply to: [329039442](https://github.com/dotnet/coreclr/pull/26918#discussion_r329039442) [](ancestors = 329039442)